lib/forth_evaluator/dictionary.ex

```diff
@@ -35,50 +37,41 @@ defmodule ForthEvaluator.Dictionary do
     Agent.stop(dictionary)
   end
 
-  @spec store(dict :: pid(), word :: Parser.word(), tokens :: [Parser.token()]) :: no_return()
+  @spec store(dict :: pid(), word :: Parser.word(), tokens :: [Parser.token()]) :: op_result
   @doc """
   Stores the given tokens under the given name into the dictionary.
 
-    iex> {:ok, dictionary} = ForthEvaluator.Dictionary.start_link()
+    iex> {:ok, dictionary} = start_link()
     iex> Agent.get(dictionary, &Function.identity/1)
     %{}
-    iex> ForthEvaluator.Dictionary.store(dictionary, "word", [1, 2])
+    iex> store(dictionary, "word", [1, 2])
     iex> Agent.get(dictionary, &Function.identity/1)
     %{"word" => [1, 2]}
 
-    iex> {:ok, dictionary} = ForthEvaluator.Dictionary.start_link(%{"word1" => [1]})
+    iex> {:ok, dictionary} = start_link(%{"word1" => [1]})
     iex> Agent.get(dictionary, &Function.identity/1)
     %{"word1" => [1]}
-    iex> ForthEvaluator.Dictionary.store(dictionary, "word2", [2, 3])
+    iex> store(dictionary, "word2", [2, 3])
     iex> Agent.get(dictionary, &Function.identity/1)
     %{"word1" => [1], "word2" => [2, 3]}
 
   """
   def store(dictionary, word, tokens) do
     Agent.update(dictionary, fn state -> Map.put(state, word, tokens) end)
+    {:ok, ""}
   end
 
-  @spec search(dict :: pid(), word :: Parser.word()) :: [Parser.token()] | :unknown
+  @spec search(dict :: pid(), stack :: pid(), word :: Parser.word()) :: [op_result]
   @doc """
-  Retrieves the tokens stored under the given word.
-  If the word is not in the dictionary, returns the `:unknown` atom.
-
-    iex> {:ok, dictionary} = ForthEvaluator.Dictionary.start_link()
-    iex> ForthEvaluator.Dictionary.store(dictionary, "word", [1, 2])
-    iex> ForthEvaluator.Dictionary.search(dictionary, "word")
-    [1, 2]
-
-    iex> {:ok, dictionary} = ForthEvaluator.Dictionary.start_link(%{"word1" => [1]})
-    iex> ForthEvaluator.Dictionary.store(dictionary, "word2", [2, 3])
-    iex> ForthEvaluator.Dictionary.search(dictionary, "word2")
-    [2, 3]
-
-    iex> {:ok, dictionary} = ForthEvaluator.Dictionary.start_link()
-    iex> ForthEvaluator.Dictionary.search(dictionary, "unknown_word")
-    :unknown
-
+  Executes the tokens asociated with a given word and returns their results as a list.
+  If the word is not found, returns an error.
   """
-  def search(dictionary, word) do
-    Agent.get(dictionary, fn state -> Map.get(state, word, :unknown) end)
+  def search(dictionary, stack, word) do
+    Agent.get(dictionary, fn state ->
+      case Map.get(state, word, nil) do
```

Comment:
```suggestion
      case Map.get(state, word) do
```

Map.get/2 default return value is `nil` 😄